chia/full_node/full_node.py

```diff
@@ -595,7 +595,9 @@ async def short_sync_batch(self, peer: WSChiaConnection, start_height: uint32, t
                 self.sync_store.batch_syncing.remove(peer.peer_node_id)
                 self.log.error(f"Error short batch syncing, could not fetch block at height {start_height}")
                 return False
-            if not self.blockchain.contains_block(first.block.prev_header_hash):
+            hash = self.blockchain.height_to_hash(first.block.height - 1)
+            assert hash is not None
+            if not hash == first.block.prev_header_hash:
```

Comment:
Why not do this if hash is None instead of the assert

Reply:
because we are guaranteed to have this height

Reply:
is `not A == B` better than `A != B`? It seems a bit more convoluted.